### PR TITLE
Extract code for classifying a document span into its own helper class.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Extract code for classifying a document span into its own helper class.

This is to help the https://github.com/dotnet/roslyn/pull/27818 PR.  

I recommend reading this one commit at a time.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @JieCarolHu .  This is what i was envisioning.  After this PR you can just call ```await EditorClassifier.GetClassifiedSpansAsync```.  You just need to check if that returns 'IsDefault' or not.

File: src/EditorFeatures/Core/FindUsages/ClassifiedSpansAndHighlightSpanFactory.cs [hunk 1]
```diff
@@ -100,194 +91,5 @@ private static TextSpan GetLineSpanForReference(SourceText sourceText, TextSpan
                 new ClassifiedSpan(ClassificationTypeNames.Text, narrowSpan),
                 new ClassifiedSpan(ClassificationTypeNames.Text, TextSpan.FromBounds(narrowSpan.End, widenedSpan.End)));
         }
-
-
-        private static async Task<ImmutableArray<ClassifiedSpan>> GetClassifiedSpansAsync<TClassificationService>(
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/FindUsages/ClassifiedSpansAndHighlightSpanFactory.cs]: All this code moved, unchanged, into EditorClassifier.

File: src/EditorFeatures/Core/FindUsages/ClassifiedSpansAndHighlightSpanFactory.cs [hunk 2]
```diff
@@ -100,12 +95,43 @@ private static TextSpan GetLineSpanForReference(SourceText sourceText, TextSpan
                 new ClassifiedSpan(ClassificationTypeNames.Text, narrowSpan),
                 new ClassifiedSpan(ClassificationTypeNames.Text, TextSpan.FromBounds(narrowSpan.End, widenedSpan.End)));
         }
+    }
+
+
+    internal static class EditorClassifier
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/FindUsages/ClassifiedSpansAndHighlightSpanFactory.cs]: this commit just extracts the code above into EditorClassifier.  The next commit moves it into its own file.

File: src/EditorFeatures/Core/FindUsages/ClassifiedSpansAndHighlightSpanFactory.cs [hunk 3]
```diff
@@ -96,224 +92,4 @@ private static TextSpan GetLineSpanForReference(SourceText sourceText, TextSpan
                 new ClassifiedSpan(ClassificationTypeNames.Text, TextSpan.FromBounds(narrowSpan.End, widenedSpan.End)));
         }
     }
-
-
-    internal static class EditorClassifier
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/FindUsages/ClassifiedSpansAndHighlightSpanFactory.cs]: this commit only moves this file.  no other changes happen.

**Comment by CyrusNajmabadi** [comment]
tagging @dotnet/roslyn-ide

**Comment by heejaechang** [comment]
curious. what would be the difference between this and classifier? if they are same, can they share code? also, this is also static, won't it have the same issue as classifier having an issue with non-C#/VB language?

**Comment by CyrusNajmabadi** [comment]
> curious. what would be the difference between this and classifier? 

This is a static entrypoint that first tries to use the new IClassifierService architecture.  If that is missing, it falls back to the older, deprecated, IEditorClassifierService architecture.

> if they are same, can they share code? 

If a language exports an IClassifierService, this will call into that.  So they are sharing code in that sense.

> also, this is also static, won't it have the same issue as classifier having an issue with non-C#/VB language?

No.  This ends up querying for the appropriate ILanguageService.  So this can work with any Roslyn language.  This is already the system that is used by the part of FindReferences that produces classified text runs.  So it's already works for C#/Vb/TS/JS.  Since htat's what we want for QuickInfo now, i'm pulling htis out into a better location so it can be more easily used by @JieCarolHu .

**Comment by JieCarolHu** [comment]
Merged into in the close brace PR.  Thanks! @CyrusNajmabadi

**Pull request closed without merging**